datafusion/expr/src/built_in_function.rs

```diff
@@ -464,12 +479,16 @@ impl BuiltinScalarFunction {
             },
             BuiltinScalarFunction::ArrayConcat => {
                 let mut expr_type = Null;
+                let mut max_dims = 1;
                 for input_expr_type in input_expr_types {
                     match input_expr_type {
                         List(field) => {
                             if !field.data_type().equals_datatype(&Null) {
```

Comment:
Is it possible to get a nested list array with a null type? i.e. List(List(Null))